### ISSUE TITLE
Discuss `RootObject.name` attribute

### ISSUE THREAD

**Issue by steffres** [acdh-oeaw/apis-core-rdf#79]
We had discussed this before, but I'm in favor of keeping one field on RootObject which acts as a **single human textual representation** of an instance. 

Whether it's in debugging, in an overview table in the UI, in REST API results, etc,anywhere I'm certain that such a single textual representation comes in handy often enough to justify this field to exist on everything in an ontology (as everything inherits from RootObject).

Maybe renaming it from `name` to `label` or `value`?

I think of two arguments for it:

### if other naming fields are used, make the `name` dependent on them

The main argument against `name` in the JF discussion to my memory was that there are entities that don't need it / or where it would lead to duplicates. With person names for examples. There it is often the case that they have a surname and a last name, where an addition name field could lead to confusion. In such a case I'd say hide `RootObject.name` from the user edit UI, and add `surname` and `lastname` fields. Whenever `save` function is called, a combination of them is persisted into `name`. 

Basically whenever `name` (or whatever we would call it) can not satisfy an entity class because more complex fields are needed, only the complex fields should be presented to the user and not the `name`, while in the background `name` will be calculated by depending on these other fields.

### usage for property management

Also it could serve as part of a solution to https://github.com/acdh-oeaw/apis-core-rdf/issues/59 where we need a single identifying attribute to manage the various modifications on a property instance (since it's a django model instance and not a model class itself as entities are for example, which are picked up by Django's migration logic).

Perhaps for vocabularies too, where I forsee a similiar issue.

**Comment by steffres** [comment]
one (among many) relevant code part: https://github.com/acdh-oeaw/apis-core-rdf/blob/eb0d4bc80635d55e9ba7aa000ff614a0759c0f3a/apis_core/apis_relations/models.py#L78

**Comment by koeaw** [comment]
Commenting just to include personal input on naming suggestions (not the greater question/issue):

I would not choose `value` for field name due to the word being commonly used in a different context (key/value pair), so much so it's often even used as temporary variable in code. Would consider this an anti-pattern.

I'd suggest to use something overly explicit that a dev is unlikely to want to pick themselves (which makes `label` no different from `name`, IMO). IIRC I put forward `original_*` or `initial_*`, but a word that'd make the field's _intention_ even clearer would be better.

**Comment by steffres** [comment]
I suggest `primary_label`, since it's self-explanatory and unique enough to not conflict with other attributes of an arbitrary ontology.

**Comment by steffres** [comment]
It was decied to drop the `name` attribute entirely and not replace it with anything.
